Sources/Paywalls/Components/PaywallCarouselComponent.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallCarouselComponent.swift
+//
+//  Created by Josh Holtz on 1/26/25.
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct AutoAdvanceSlides: PaywallComponentBase {
+
+        public let msTimePerSlide: Int
+        public let msTransitionTime: Int
+
+        public init(msTimePerSlide: Int, msTransitionTime: Int) {
+            self.msTimePerSlide = msTimePerSlide
+            self.msTransitionTime = msTransitionTime
+        }
+
+    }
+
+    final class CarouselComponent: PaywallComponentBase {
+
+        let type: ComponentType
+        public let alignment: VerticalAlignment
+        public let size: Size
+        public let spacing: CGFloat?
+        public let padding: Padding
+        public let margin: Padding
+        public let shape: Shape?
+        public let border: Border?
+        public let shadow: Shadow?
```

Comment:
Should we also add a background color property? Also do we need to add the page indicator settings here?

Reply:
Another property that we might want is the initialPage to use